src/soter/openssl/soter_sign_ecdsa.c

```diff
@@ -102,7 +102,7 @@ soter_status_t soter_sign_update_ecdsa_none_pkcs8(soter_sign_ctx_t* ctx, const v
 soter_status_t soter_sign_final_ecdsa_none_pkcs8(soter_sign_ctx_t* ctx, void* signature, size_t *signature_length)
 {
   EVP_PKEY *pkey = EVP_PKEY_CTX_get0_pkey(ctx->pkey_ctx);
-  if (!pkey && EVP_PKEY_base_id(pkey)!=EVP_PKEY_EC){
+  if (!pkey || EVP_PKEY_base_id(pkey)!=EVP_PKEY_EC){
```

Comment:
You are correct - thank you. I would also try to improve it a bit by not relying on potential compiler "optimisations" in evaluating this statement and split it into two consecutive ones, like
```
if (!pkey) {
  return SOTER_INVALID_PARAMETER;
}
if (EVP_PKEY_base_id(pkey)!=EVP_PKEY_EC) {
  return SOTER_INVALID_PARAMETER;
}
```

this makes it more explicit and readable